File: binding/src/charts/libqpolarchart.cpp
#include <QAbstractAxis>
#include <QAbstractSeries>
#include <QChart>
#include <QGraphicsItem>
#include <QPolarChart>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qpolarchart.h>
#include "libqpolarchart.h"
#include "libqpolarchart.hxx"

QPolarChart* QPolarChart_new() {
	 return new QPolarChart();
}

QPolarChart* QPolarChart_new2(QGraphicsItem* parent) {
	 return new QPolarChart(parent);
}

QPolarChart* QPolarChart_new3(QGraphicsItem* parent, int wFlags) {
	 return new QPolarChart(parent, static_cast<QFlags<Qt::WindowType>>(wFlags));
}

libqt_string QPolarChart_Tr(const char* s) {
	QString _ret = QPolarChart::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QPolarChart_AddAxis(QPolarChart* self, QAbstractAxis* axis, int polarOrientation) {
	self->addAxis(axis, static_cast<QPolarChart::PolarOrientation>(polarOrientation));
}

libqt_list QPolarChart_Axes(const QPolarChart* self) {
	auto _ret = self->axes();
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		_data[_i] = _ret[_i];
	}
	return _arr;
}

int QPolarChart_AxisPolarOrientation(QAbstractAxis* axis) {
	return QPolarChart::axisPolarOrientation(axis);
}

libqt_string QPolarChart_Tr2(const char* s, const char* c) {
	QString _ret = QPolarChart::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QPolarChart_Tr3(const char* s, const char* c, int n) {
	QString _ret = QPolarChart::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_list QPolarChart_Axes1(const QPolarChart* self, int polarOrientation) {
	auto _ret = self->axes(static_cast<QFlags<QPolarChart::PolarOrientation>>(polarOrientation));
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		_data[_i] = _ret[_i];
	}
	return _arr;
}

libqt_list QPolarChart_Axes2(const QPolarChart* self, int polarOrientation, QAbstractSeries* series) {
	auto _ret = self->axes(static_cast<QFlags<QPolarChart::PolarOrientation>>(polarOrientation), series);
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		_data[_i] = _ret[_i];
	}
	return _arr;
}

void QPolarChart_Delete(QPolarChart* self) {
    delete self;
}

File: binding/src/charts/libqpolarchart.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qpolarchart.html
pub const qpolarchart = struct {

    /// New constructs a new QPolarChart object.
    pub fn New() QtC.QPolarChart {
        return qtc.QPolarChart_new();
    }


    /// New2 constructs a new QPolarChart object.
    pub fn New2(parent: ?*anyopaque) QtC.QPolarChart {
        return qtc.QPolarChart_new2(@ptrCast(parent));
    }


    /// New3 constructs a new QPolarChart object.
    pub fn New3(parent: ?*anyopaque, wFlags: i64) QtC.QPolarChart {
        return qtc.QPolarChart_new3(@ptrCast(parent), @intCast(wFlags));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPolarChart_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpolarchart.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn AddAxis(self: ?*anyopaque, axis: ?*anyopaque, polarOrientation: i32) void {
        qtc.QPolarChart_AddAxis(@ptrCast(self), @ptrCast(axis), @intCast(polarOrientation));
    }

    pub fn Axes(self: ?*anyopaque, ) []const u8 {
        return qtc.QPolarChart_Axes(@ptrCast(self));
    }

    pub fn AxisPolarOrientation(axis: ?*anyopaque) i32 {
        return qtc.QPolarChart_AxisPolarOrientation(@ptrCast(axis));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPolarChart_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpolarchart.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPolarChart_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpolarchart.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Axes1(self: ?*anyopaque, polarOrientation: i32) []const u8 {
        return qtc.QPolarChart_Axes1(@ptrCast(self), @intCast(polarOrientation));
    }

    pub fn Axes2(self: ?*anyopaque, polarOrientation: i32, series: ?*anyopaque) []const u8 {
        return qtc.QPolarChart_Axes2(@ptrCast(self), @intCast(polarOrientation), @ptrCast(series));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPolarChart_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/polarchart.html#types
pub const enums = struct {
    pub const PolarOrientation = enum {
        pub const PolarOrientationRadial: i32 = 1;
        pub const PolarOrientationAngular: i32 = 2;
    };

};
